src/algorithms.cpp

```diff
@@ -22,4 +24,105 @@ bool isComplete(const Model &model) {
     return true;
 }
 
+bool isLocCorrect(const Model& model, IdRef id){
+	const Node* node = model.node(id);
+
+	auto isInward = std::vector<bool>(Node::MAX_SLOTS, false);
+	auto isOutward = std::vector<bool>(Node::MAX_SLOTS, false);
+
+
+	if (node->type()==END) {
+		// Single slot of end must be connected to bidir. section or forbidden section
+		return model.section(node->section(0))->dir()!=Section::AllowedTravel::UNIDIR;
+	}
+
+	else{
+		// Set outward/inward flags
+		for(SlotId start = 0; start < node->sectionCount(); ++start){
+			for(SlotId end = 0; end < node->sectionCount(); ++end){
+				if(
```

Comment:
I think this could be made more readable if you use boolean flags instead of packing everything inside the if condition

Reply:
Changed in a6b9f90c3de171f5e65f18c54d3cfd37a8e4bacd